src/catchup/CatchupManagerImpl.cpp

```diff
@@ -93,4 +217,80 @@ CatchupManagerImpl::logAndUpdateCatchupStatus(bool contiguous)
 {
     logAndUpdateCatchupStatus(contiguous, getStatus());
 }
+
+bool
+CatchupManagerImpl::hasBufferedLedger() const
+{
+    return !mSyncingLedgers.empty();
+}
+
+LedgerCloseData const&
+CatchupManagerImpl::getBufferedLedger() const
+{
+    if (!hasBufferedLedger())
+    {
+        throw std::runtime_error(
+            "getBufferedLedger called when mSyncingLedgers is empty!");
+    }
+
+    return mSyncingLedgers.front();
+}
+
+void
+CatchupManagerImpl::popBufferedLedger()
+{
+    if (!hasBufferedLedger())
+    {
+        throw std::runtime_error(
+            "popBufferedLedger called when mSyncingLedgers is empty!");
+    }
+
+    mSyncingLedgers.pop_front();
+    mSyncingLedgersSize.set_count(mSyncingLedgers.size());
+}
+
+void
+CatchupManagerImpl::syncMetrics()
+{
+    mSyncingLedgersSize.set_count(mSyncingLedgers.size());
+}
+
+void
+CatchupManagerImpl::reset()
+{
+    mCatchupWork.reset();
+    resetSyncingLedgers();
+}
+
+void
+CatchupManagerImpl::resetSyncingLedgers()
+{
+    mSyncingLedgers.clear();
```

Comment:
I don't know if you need this method if all it does is clear `mSyncingLedgers` (see my other comment on metrics)

Reply:
removed